content-src/components/StartupOverlay/StartupOverlay.jsx

```diff
@@ -12,11 +12,24 @@ export class _StartupOverlay extends React.PureComponent {
     this.initScene = this.initScene.bind(this);
     this.removeOverlay = this.removeOverlay.bind(this);
     this.onInputInvalid = this.onInputInvalid.bind(this);
+    this.setFxaMetrics = this.setFxaMetrics.bind(this);
 
     this.state = {
       emailInput: "",
-      overlayRemoved: false
+      overlayRemoved: false,
+      flowId: "",
+      flowBeginTime: ""
     };
+  }
+
+  componentWillUpdate() {
+    if (this.props.fxa_endpoint && !this.didFetch) {
+      this.didFetch = true;
+      this.setFxaMetrics();
```

Comment:
do you need to `await` on this function here since it's async?

Reply:
if you remove this function and just put the logic in componentWillUpdate then you can disregard this comment

Reply:
I was going to put the logic in the componentWillUpdate, but then it runs multiple times being async, the flag does not get set in time to stop the second time from running. 
I was also considering, on failure, doing a setTimout and running this again up to 2 or 3 times (circumventing the flag) just in case there is a temporary error, like a bad connection or something.